src/WingetCreateCore/Common/GitHub.cs

```diff
@@ -59,6 +59,18 @@ public static async Task<string> GetGitHubAppInstallationAccessToken(string gitH
             var installation = await github.GitHubApps.GetRepositoryInstallationForCurrent(wingetRepoOwner, wingetRepo);
             var response = await github.GitHubApps.CreateInstallationToken(installation.Id);
             return response.Token;
+        }
+
+        /// <summary>
+        /// Gets the latest release tag name of winget-create.
+        /// </summary>
+        /// <returns>Latest release tag name.</returns>
+        public static async Task<string> GetLatestRelease()
+        {
+            var github = new GitHubClient(new ProductHeaderValue(UserAgentName));

```

Comment:
This is going to hit [https://github.com/microsoft/winget-create/issues/116](https://github.com/microsoft/winget-create/issues/116), potentially at the beginning of every launch. This is tricky, since this may run for a command that doesn't even have the token parameter.

I'd say at minimum use the user's token parameter if it exists, and when you do your refactor, use the cached token as well. 

OR

If this fails due to rate limiting, just ignore that failure

Reply:
I hadn't thought about rate limits when checking for release. I will probably hold off on this PR until the token caching refactor is complete.

Reply:
I have changed the logic to be similar to what you suggested. The way it works now is that if a token is present (either as an argument from cache) it will use that to get the latest version. If not, it will still try to get the latest version, and any rate limiting failures are caught and ignored.